chore: hide unavailable bounties

Keyrxng opened the pull request:
Resolves https://github.com/ubiquity/devpool-directory-bounties/issues/11

- https://github.com/Keyrxng/devpool-directory/issues
- https://github.com/Keyrxng/devpool-directory/issues/34

I'm currently assigned the dynamic rpc handler task so in my forked devpool repo it's closed

Keyrxng committed:
chore: hide unavailable bounties

0x4007 commented:
@Keyrxng as a heads up I'm sort of stepping away from code reviews these days, so that I can spend more time on recruiting and fundraising. In the short term I plan to just stay on top of conversations at a high level in order to remedy any operational blockers during this transition phase. 

In this situation I think you should request for reviews until we figure out who will be the de-facto "manager" of this repository, although I would default to rndquu if I'm not sure for most cases.

Keyrxng commented:
I hope that's a good thing for you but cheers mate I'll ping rndquu or paul from now on

rndquu commented on index.ts:
```diff
@@ -132,7 +157,8 @@ async function main() {
           if (!(projectIssue.labels as GitHubLabel[]).some((label) => label.name.includes(LABELS.PRICE))) continue;
-
+          // if issue has the "Unavailable" label then skip it, no need to pollute repo with new issues that are already taken
+          if (!(projectIssue.labels as GitHubLabel[]).some((label) => label.name.includes(LABELS.UNAVAILABLE))) continue;
```
Project (i.e. partner) issues don't have the `Unavailable` tag, only devpool issues from https://github.com/ubiquity/devpool-directory/issues

Here we should check that "if project issue is assigned to somebody then skip such issue", no need to check the `Unavailable` label

rndquu commented on index.ts:
```diff
@@ -99,6 +99,31 @@ async function main() {
           const projectIssueLabelsStringified = getDevpoolIssueLabels(projectIssue, projectUrl).sort().toString();
+
+          // if bounty is unavailable and exists in the list with the state open then close such bounty
+          if (isDevpoolUnavailableLabel && devpoolIssue.state === "open") {
```
No need to check the `Unavailable` label

If project (i.e. partner) issue is assigned to somebody and devpool issue from https://github.com/ubiquity/devpool-directory/issues is "open" then close such issue

rndquu commented on index.ts:
```diff
@@ -99,6 +99,31 @@ async function main() {
+          }
+
+          // if bounty is available exists in the list with the state closed then reopen such bounty
+          if (!isDevpoolUnavailableLabel && devpoolIssue.state === "closed") {
```
Again, no need to check the `Unavailable` label

If project (i.e. partner) issue is NOT assigned to anybody and devpool issue from https://github.com/ubiquity/devpool-directory/issues is "closed" then open such issue

rndquu reviewed (changes requested):
Overall, when this PR is merged we're free to remove the "Unavailable label" feature since all of the open issues in the devpool will be available for self-assignment

Keyrxng committed:
Update total rewards

Keyrxng committed:
fix: requested changes

Keyrxng committed:
Merge branch 'ubiquity:development' into development

Keyrxng commented:
https://github.com/Keyrxng/devpool-directory/issues/32

0x4007 commented on total-rewards.txt:
```diff
@@ -1 +1 @@
-11573
\ No newline at end of file
+2787
```
I just realized that if we remove the assigned tasks, then it messes with the stats. How can we preserve the stats for marketing purposes? It would be great to write a JSON object with the following:
```typescript
type Rewards = {
  notAssigned: number;
  assigned: number;
  completed: number;
  total: number;
}
```

Also potentially useful stats:
```typescript
type Tasks = {
  notAssigned: number;
  assigned: number;
  completed: number;
  total: number;
}
```

Perhaps this feature needs to be reimplemented? Its already in use on ubq.fi and I have ideas for how to use it on Twitter too.

rndquu replied:
Why do we need to keep track of total rewards in a separate file? All of the tasks/rewards data can be fetched via github API in a single script.

0x4007 replied:
Caching seems appropriate vs paginating tens of pages of results for our index at ubq.fi. 

Given that it's loaded below the fold it's probably fine though.

rndquu replied:
I'm saying that it's simpler to create a separate script that fetches/syncs ubiquity's github organization state (issues, comments, etc...) and saves it to supabase (which we can use later to fetch data for analytics UI) compared to keeping some json file in each repository.

0x4007 replied:
> keeping some json file in each repository.

No we just need to store it in the `devpool-directory` repository. I don't see a reason for why we would need to cache within every repository. This data is intended to be system wide statistics, primarily for marketing purposes, which the cron job already has to process as it updates the mirrored issues states.

rndquu commented:
@Keyrxng Pls notice that requested changes are not implemented

Keyrxng committed:
Merge branch 'ubiquity:development' into development

Keyrxng committed:
fix: requested changes

Keyrxng commented:
> @Keyrxng Pls notice that requested changes are not implemented

I was thinking wth I did make the changes, but after checking I realize I forgot to push lmao

rndquu commented on index.ts:
```diff
@@ -92,13 +92,39 @@ async function main() {
+          }
+
+          // if the project issue is not assigned to someone and it's closed in the devpool, then reopen it
+          if (!projectIssue.assignee?.login && devpoolIssue.state === "closed") {
```
If project issue is closed as "not planned" without an assignee then such issue will be reopened in the devpool directory while the expected behaviour is to keep it closed.

I guess here we should also check that `projectIssue.state === "open"`

Example:
Closed project issue without assignee: https://github.com/rndquu-org/test-repo/issues/51
Reopened issue: https://github.com/rndquu/devpool-directory/issues/64

Keyrxng committed:
Merge branch 'ubiquity:development' into development

Keyrxng committed:
fix: check partner issue is open too

rndquu reviewed (approved):
Works fine

Keyrxng commented:
@rndquu should the checks against the unavailable labels be removed? I think that's what's causing the open/close cycle that's happening right now. 

Looking at the logs 

second most recent run:
```
Closed (assigned): https://github.com/ubiquity/devpool-directory/issues/1125 (https://github.com/ubiquity/pay.ubq.fi/issues/196)
```

most recent run:
```
Updated: https://github.com/ubiquity/devpool-directory/issues/1125 (https://github.com/ubiquity/pay.ubq.fi/issues/196)
```

with `"Updated: ..."` it's happening outside of the two conditionals I added, and you said that there's no need to check those labels and they'll be getting removed anyway?

- https://github.com/ubiquity/devpool-directory/blob/a73a14f7fb5e9c51392d874bfd98d2232a1fce1a/index.ts#L139

0x4007 commented:
@Keyrxng test on your fork then to prove that its a solution. I would consider this problem urgent because it negatively affects new people coming to work.ubq.fi to see if they want to work with us.

rndquu commented:
> @rndquu should the checks against the unavailable labels be removed?

Yes

> with "Updated: ..." it's happening outside of the two conditionals I added, and you said that there's no need to check those labels and they'll be getting removed anyway?

Yes

Related issue https://github.com/ubiquity/devpool-directory-bounties/issues/22

Keyrxng commented:
Only just seeing this now, my notifications got tanked after I chose only the day before yesterday to subscribe to devpool notifications lmao. I'll test it before pushing
